### PR TITLE
fix(terminal): launch prior/default agent when a woken terminal can't resume

### LINKED ISSUE #4557: [Bug]: Awaking sleeping session does not start agent automatically
### Operating system

macOS

### Orca version

_No response_

### Details

So I'm not sure if this is actually a bug or if it's meant to do it like this. Basically, if a worktree is asleep and you wake it, instead of starting with a terminal with your default agent, it starts on a blank terminal. You would have to call Claude or whoever it is to properly start it up again. It will be nice if, when you actually wake it, it does the same as if you're creating a new worktree, where it would just create this on your default agent that you can set

### PR DESCRIPTION
**Pull request by gatsby74**
## Summary

Waking a sleeping worktree — or restoring terminals after an app restart — left a terminal that previously ran an agent as a **blank shell** while the tab still showed the agent's icon/title. You had to manually re-run `claude`/`codex` to get going.

Root cause: in `pty-connection.ts`, `buildColdRestoreAgentResumeStartup()` returns `null` whenever the agent's provider session can't be resumed (no provider session, or the resume plan can't be built). The caller then spawns a bare shell, and nothing clears the tab's `launchAgent`, so the stale agent icon remains.

This change adds a fresh-launch fallback at exactly those `null` points. When resume isn't possible, an **agent terminal** now comes back as a **fresh session** (no `--resume`) of the agent it previously ran, resolved in priority order:

1. live agent status (`agentType`), else
2. the sleeping record's agent, else
3. the terminal's persisted `tab.launchAgent` (the durable signal across restarts), else
4. the pane's startup `launchAgent`.

If the prior agent is unknown **but there's evidence the terminal was an agent terminal** (a live status entry or sleeping record exists), the configured **default agent** is used — exactly like opening a new terminal. A **genuine plain shell** (no agent ever ran there, no default trigger) stays blank. No context is carried over (a clean session), and the correct agent icon is restored because the fresh startup flows through the existing cold-restore path that sets `launchAgent`.

Fixes #4557

## Screenshots

Waking a slept agent terminal — before vs after (PII redacted):

https://github.com/user-attachments/assets/56e7a654-6699-48dc-a68b-cdab23e9864e

- **Before**: the woken terminal comes up as a blank shell (with a stale agent icon).
- **After**: it relaunches the agent it previously ran; a plain shell still wakes blank.

## Testing

- [x] `pnpm lint`
- [x] `pnpm typecheck`
- [x] `pnpm test` — `pty-connection.test.ts` passes (193 incl. 4 new cases). The only full-suite failures are pre-existing, environment-only flakes unrelated to this change (`src/relay/git-handler.test.ts`, `src/relay/subprocess.test.ts` — they pass in isolation and flake under parallel load; they also fail on a clean `upstream/main`).
- [x] `pnpm build`
- [x] Added regression tests: unresumable sleeping record → fresh agent; persisted `tab.launchAgent` → fresh prior agent after restart; agent terminal with unrecoverable agent → default agent; plain shell stays blank.

## AI Review Report

Ran an adversarial review of the diff. Risks checked and outcomes:

- **Forcing an agent into a plain shell / new terminal** — verified safe: the `pendingStartupCommand` guard returns null for any launch that already carries a startup command; the default-agent fallback only fires when a live entry or sleeping record exists; a brand-new blank terminal has none of those (even with a default agent configured), so it stays blank. New-terminal creation does not route through this builder.
- **Resume behavior unchanged** — the fallback is only reached at the three sites that previously returned `null`; the successful-resume branch is untouched (existing resume tests pass).
- **Cross-platform (macOS/Linux/Windows + WSL)** — the fresh launch uses the same `getColdRestoreAgentResumePlatform()` and `buildAgentStartupPlan` (same shell resolution/quoting) as the resume path; empty prompt avoids quoting risk; WSL→`linux` mapping is identical. No shortcuts/labels/paths touched.
- **SSH/remote** — parity confirmed: the fresh startup object has the same shape as the resume object and is delivered through the existing SSH cold-restore mechanism; no SSH-specific code change.
- **Type widening** — `ColdRestoreAgentResumeStartup.agent` widened from `ResumableTuiAgent` to `TuiAgent` (a fresh launch can target a non-resumable agent). All consumers accept `TuiAgent` (`registerAgentLaunchConfig` metadata is `AgentType`; `launchAgent` fields are `TuiAgent`).
- Hardened `clearSleepingRecordProviderDuplicates` to tolerate a record without a `providerSession` (now reachable via the fresh-launch path).

## Security Audit

- **Command execution** — the fresh launch command is built by the same `buildAgentStartupPlan` used for new terminals; no user-derived strings are interpolated unquoted (empty prompt), and args/env come from the existing settings-resolution helpers.
- **No** new IPC surface, secrets, auth, path handling, or dependency changes.

## Notes

- Behavior is gated so plain shells and blank new terminals are never converted into agents.
- The fresh launch starts a clean session (no transcript continuation), matching the agreed behavior.

### CODE REVIEW

**Comment by mmarabel** [comment]
walkthrough 

https://github.com/user-attachments/assets/56e7a654-6699-48dc-a68b-cdab23e9864e

File: src/renderer/src/components/terminal-pane/pty-connection.test.ts [hunk 1]
```diff
@@ -3807,6 +3810,128 @@ describe('connectPanePty', () => {
...
+      sleepingAgentSessionsByPaneKey: {
+        [paneKey]: { paneKey, tabId: 'tab-1', worktreeId: 'wt-1', capturedAt: 1, updatedAt: 1 }
+      }
+    } as StoreState
+
+    await mountColdRestore(transport)
+
+    const reattach = findReattachConnect(transport)
+    expect(reattach?.command).toMatch(/^codex /)
+    expect(reattach?.launchAgent).toBe('codex')
```
[the remark this answers is not in the document]
Comment (gatsby74) [reply src/renderer/src/components/terminal-pane/pty-connection.test.ts:3998]: Done in 0907fd37c — added `expect(reattach?.command).not.toContain('resume')` to the default-agent fallback test to enforce the fresh-launch contract. All 193 tests pass.

**Review by Jinwoo-H — APPROVED** [review]
Thanks for the fix. I added a small maintainer follow-up to cover the cold-restore edge cases:

- Fresh fallback ignores stale completed live rows.
- Stored launch config/env is preserved when relaunching the agent.
- SSH fallback uses the known remote platform.
- Malformed sleeping records are handled defensively.

Validated with focused terminal-pane tests, typecheck, lint, and Electron smoke coverage.
